python/lsst/summit/utils/nightReport.py

```diff
@@ -0,0 +1,633 @@
+
+        Parameters
+        ----------
+        threshold : `float`, optional
+            The minimum time gap to print out, in seconds.
+        includeCalibs : `bool`, optional
+            If True, start at the lowest seqNum, otherwise start when the
+            night's observing started.
+        """
+        if not HAVE_HUMANIZE:
+            self.log.warning('Please install humanize to use make this print as intended.')
+        dts = self.getTimeDeltas()
+
+        allSeqNums = list(self.data.keys())
+        if includeCalibs:
+            seqNums = allSeqNums
+        else:
+            firstObs = self.getObservingStartSeqNum(method='safe')
+            # there is always a big gap before firstObs by definition so add 1
+            startPoint = allSeqNums.index(firstObs) + 1
+            seqNums = allSeqNums[startPoint:]
+
+        messages = []
+        for seqNum in seqNums:
+            dt = dts[seqNum]
+            if dt > threshold:
+                messages.append(f"seqNum {seqNum:3}: {precisedelta(dt)} gap")
+
+        if messages:
+            print(f"Gaps between observations greater than {threshold}s:")
+            for line in messages:
+                print(line)
+
+    def getObservingStartSeqNum(self, method='safe'):
+        """Get the seqNum at which on-sky observations started.
+
+        If no on-sky observations were taken ``None`` is returned.
+
+        Parameters
+        ----------
+        method : `str`
+            The calculation method to use. Options are:
+            - 'safe': Use the first seqNum with an observation_type that is
+            explicitly not a calibration or test. This is a safe way of
+            excluding the calibs, but will include observations where we
+            take some closed dome test images, or start observing too early,
+            and fo back to taking calibs for a while before the night starts.
+            - 'heuristic': Use a heuristic to find the first seqNum. The
+            current heuristic is to find the first seqNum with an observation
+            type of CWFS, as we always do a CWFS focus before going on sky.
+            This does not work well for old days, because this wasn't always
+            the way data was taken. Note: may be updated in the future, at
+            which point this will be renamed ``cwfs``.
+
+        Returns
+        -------
+        startSeqNum : `int`
+            The seqNum of the start of the night's observing.
+        """
+        allowedMethods = ['heuristic', 'safe']
+        if method not in allowedMethods:
+            raise ValueError(f"Method must be one of {allowedMethods}")
+
+        if method == 'safe':
+            # as of 20221211, the full set of observation_types ever seen is:
+            # acq, bias, cwfs, dark, engtest, flat, focus, science, stuttered,
+            # test, unknown
+            offSkyObsTypes = ['bias', 'dark', 'flat', 'test', 'unknown']
```

Comment:
Is this likely to change and be irritating to keep updated?

Reply:
I hope not! (It's based on me scraping almost 3 years of data now). But if it is, then there's like not much that can be done, I think it has to be human-curated. I think it's fine for now though.